Id 544 (#52)
xebialabs/xl-docker-images: docs/environment-variables.md

- example: "adm_P@ssw0rd_01"

##### `REPOSITORY_KEYSTORE_PASSPHRASE`
- Repository keystore passphrase used in XL Products.
- possible values: "passphrase-string-value"
- default value: If this value is not passed a random 16 characters password will be generated.
- example: "XL_P@ssphr@se_01"
- **Note:**
    - Keystore password must be at least 6 characters.

##### `REPOSITORY_KEYSTORE`
- Repository keystore used in XL Products.
- possible values: "keystore-string-value"
- default value: content of `${APP_HOME}/conf/repository-keystore.jceks` which is generated from `keytool -genseckey -alias deployit-passsword-key -keyalg aes -keysize 128 -keypass "deployit" -keystore ${APP_HOME}/conf/repository-keystore.jceks -storetype jceks -storepass ${REPOSITORY_KEYSTORE_PASSPHRASE}`
- example: "b3J0IHBvbGljeTogMSBtb250aCBvZiB0ZWNobmljYWwgc3VwcG9ydApFZGl0aW9uOiBUcmlhbAotLS0gU2lnbmF0dXJlIChTSEExd2l0aERTQSkgLS0tCjMwMmMwM"
- **Note:**    
    - This value is set as string whihc will be converted to base64 value and added to keystore file like below,
`echo "${REPOSITORY_KEYSTORE}" | base64 -d > ${APP_HOME}/conf/repository-keystore.jceks`

##### `XL_LICENSE`

> - Sets your XL License by passing a base64 string license, which will then be added to license file. For more details on other options to pass the license as a volume, see [Volumes](##).

Reviewer: ```suggestion
- Sets your XL License by passing a base64 string license, which will then be added to the license file. For more details on other options to pass the license in a volume, see [Volumes](##).
```

- possible values: "license-base64-value"
- default value: unregistered license valid for 7 days but you must use `ACCEPT_EULA`.
- example: "LS0tIExpY2Vuc2UgLS0tCkxpY2Vuc2UgdmVyc2lvbjogMwpQcm9kdWN0OiOiBUcmlhbAotLS0gU2lnbmF0dXJlIChTSEExd2l0aERTQSkgLS0tCjMwMmMwMjE0MTY5YjUyNGFkYWYwYWRhYmNlZDcyMGM0YmZkMGmVhNTZhNGFiMzk4YTgzNGUyZjJiOWQ1CjI2MzNhHVyZSAtLS0K%"

##### `ACCEPT_EULA`

> -"You must accept the End User License Agreement" if are unable to provide your own license before container can start.

Reviewer: ```suggestion
-"You must accept the End User License Agreement" if you are unable to provide your own license before the container can start.
```

- possible values: "y" or "Y"
- default value: none

##### `FORCE_UPGRADE`
- Force upgrade setting. It can be used to perform an upgrade in non-interactive mode by passing flag `-force-upgrades` while  starting a service.
- possible values: "true", "false"
- default value: "false"

##### `XL_CLUSTER_MODE`

> - This is to specify if HA setup is needed and specify modes of setup.

Reviewer: ```suggestion
- This is to specify if the HA setup is needed and to specify the HA mode.
```

- possible values: "default", "hot-standby", "full"
- default value: "default"

##### `XL_DB_URL`
- This is to set a DB URL that will be used for repository.
- possible values: "url-string-value"
- default value: for XLD `"jdbc:h2:file:${APP_HOME}/repository/xl-deploy"`
                 for XLR `jdbc:h2:file:${APP_HOME}/repository/xl-release`
- example: `jdbc:postgresql://postgresql:5432/XLDATABASENAME`

##### `XL_DB_USERNAME`
- Username used to connect to the database configured with XL products.
- possible values: "user-string-value"
- default value: "sa"
- example: "xl_user"

##### `XL_DB_PASSWORD`
- Password used to connect to database configured with XL products.
- possible values: "password-value"
- default value: "123"
- example: "xl_P@ssw0rd01"

##### `XL_METRICS_ENABLED`

> - Flag to expose internal and system metrics over Java Management Extensions (JMX). This is to enable the use of a monitoring systems that can read JMX data, with XL Products.

Reviewer: ```suggestion
- Flag to expose internal and system metrics over Java Management Extensions (JMX). This is to enable the use of monitoring systems that can read JMX data, with XL Products.
```
